### PR TITLE
Storybook - fixes and expands UserMilestone story

### PR DESCRIPTION
**Pull request by lmac-1**
## Changes proposed

This PR has the following changes:
1. Fixes bullet point issue on `UserMilestone` story (more info below)
2. Expands the `UserMilestone` story to include `BasicMilestone`, `MilestoneWithImage` and `BasicGoal`

The `UserMilestone` component uses a `li` element, which is wrapped in a `ul` element in parent components in LinkFree. Because of this, on Storybook, the component doesn't render correctly: 

image

You can also see this on the [deployed version of LinkFree's Storybook](https://eddiehubcommunity.github.io/LinkFree/?path=/story/components-user-usermilestone--basic) 

I fixed this by adding a decorator to the story

```
export default {
  component: UserMilestone,
  decorators: [(story) => <ul role="list">{story()}</ul>],
};
```
This wraps the component in a `ul` element and ensures that it's rendered correctly with no bullet point.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Basic Milestone:
image

Milestone with Image:
image

Basic goal: 
image

## Note to reviewers

Hope it's okay that the two changes are combined in one PR, thought it would be okay because they are all in the same file. 

I'm hoping to make some more changes to Storybook so let me know if there's anything I've missed. 

Thanks! 

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/4033"></a>

### CODE REVIEW

**Comment by lmac-1** [comment]
Eddie mentioned it was okay to directly make a pull request for Storybook small changes. I hope that's okay - let me know if not!

File: stories/components/user/UserMilestone.stories.js [hunk 1]
```diff
@@ -2,14 +2,39 @@ import UserMilestone from "../../../components/user/UserMilestone";
 
 export default {
   component: UserMilestone,
+  decorators: [(story) => <ul role="list">{story()}</ul>],
```
[the remark this answers is not in the document]
Comment (lmac-1) [reply stories/components/user/UserMilestone.stories.js:5]: On it! Will make the change now. Didn't realise you could do it like that, thanks!

[the remark this answers is not in the document]
Comment (lmac-1) [reply stories/components/user/UserMilestone.stories.js]: Change is now made in the branch.

[the remark this answers is not in the document]
Comment (dan-mba) [reply stories/components/user/UserMilestone.stories.js]: Here is the doc that shows [decorators using JSX](https://storybook.js.org/docs/7.0/react/writing-stories/decorators)

**Review by dan-mba — APPROVED** [review]
Looks good to me

File: stories/components/user/UserMilestone.stories.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (eddiejaoude) [reply stories/components/user/UserMilestone.stories.js]: oh I didn't know that either! Thank you both 👍

**Comment by eddiejaoude** [comment]
Awesome collaboration, thank you 🎉

**Pull request merged**